include/deal.II/fe/mapping_fe_field.h

```diff
@@ -649,10 +660,57 @@ class MappingFEField : public Mapping<dim, spacedim>
 
 
   // Declare other MappingFEField classes friends.
-  template <int, int, class, class>
+  template <int, int, class, class, class>
   friend class MappingFEField;
 };
 
+
+
+/**
+ * The MappingFEField is a generalization of the MappingQEulerian class, for
+ * arbitrary vector finite elements. The two main differences are that this
+ * class uses a vector of absolute positions instead of a vector of
+ * displacements, and it allows for arbitrary FiniteElement types, instead of
+ * only FE_Q.
+ *
+ * This class effectively decouples the topology from the geometry, by
+ * relegating all geometrical information to some components of a
+ * FiniteElement vector field. The components that are used for the geometry
+ * can be arbitrarily selected at construction time.
+ *
+ * The idea is to consider the Triangulation as a parameter configuration
+ * space, on which we construct an arbitrary geometrical mapping, using the
+ * instruments of the deal.II library: a vector of degrees of freedom, a
+ * DoFHandler associated to the geometry of the problem and a ComponentMask
+ * that tells us which components of the FiniteElement to use for the mapping.
+ *
+ * Typically, the DoFHandler operates on a finite element that is constructed
+ * as a system element (FESystem()) from continuous FE_Q() (for iso-parametric
+ * discretizations) or FE_Bernstein() (for iso-geometric discretizations)
+ * objects. An example is shown below:
+ *
+ * @code
+ *    const FE_Q<dim,spacedim> feq(1);
+ *    const FESystem<dim,spacedim> fesystem(feq, spacedim);
+ *    DoFHandler<dim,spacedim> dhq(triangulation);
+ *    dhq.distribute_dofs(fesystem);
+ *    const ComponentMask mask(spacedim, true);
+ *    Vector<double> eulerq(dhq.n_dofs());
+ *    // Fills the euler vector with information from the Triangulation
+ *    VectorTools::get_position_vector(dhq, eulerq, mask);
+ *    MappingFEField<dim,spacedim> map(dhq, eulerq, mask);
+ * @endcode
+ *
+ * @deprecated Use MappingFEField<dim, spacedim, VectorType> instead.
+ */
+template <int dim, int spacedim, typename VectorType, typename DoFHandlerType>
+class DEAL_II_DEPRECATED
+  MappingFEField<dim, spacedim, VectorType, DoFHandlerType>
+  : public MappingFEField<dim, spacedim, VectorType, void>
+{
+  using MappingFEField<dim, spacedim, VectorType, void>::MappingFEField;
```

Comment:
With the constructor alias, we cannot explicitly flag the constructors of the deprecated class deprecated. Should we use constructor definitions like the following instead?
```
public:
DEAL_II_DEPRECATED MappingFEField (...) : MappingFEField<dim, spacedim, VectorType, void> (...) {}
```
Is flagging constructors deprecated necessary at all?

Reply:
Isn't the use of class already signaling the warning, so we wouldn't need to specifically look at the constructor? I do not recall how exactly the warning is emitted, though.

Reply:
I just added the `hp::DoFHandler` back into `step-27` since we deprecated this class in the same way. Turns out `gcc` signals the deprecation warning, so we should be good keeping the constructor aliases.

Reply:
Agreed.